# Statistical Modeling using Sleep Health and Lifestyle Dataset
### STAT301 Group 13 Final Report
April 12th, 2025

**Contributors:** \
Yuki Matsushima #84356377 \
Masaya Tanaka #

In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


## Introduction

## Methods and Results

### a) Data

In [2]:
# read the dataset
sleep_data <- read_csv("https://raw.githubusercontent.com/yma24ma/stat301_project/refs/heads/main/Sleep_health_and_lifestyle_dataset.csv")
head(sleep_data)

Rows: 374 Columns: 13
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (5): Gender, Occupation, BMI Category, Blood Pressure, Sleep Disorder
dbl (8): Person ID, Age, Sleep Duration, Quality of Sleep, Physical Activity...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Person ID Gender Age Occupation           Sleep Duration Quality of Sleep
1 1         Male   27  Software Engineer    6.1            6               
2 2         Male   28  Doctor               6.2            6               
3 3         Male   28  Doctor               6.2            6               
4 4         Male   28  Sales Representative 5.9            4               
5 5         Male   28  Sales Representative 5.9            4               
6 6         Male   28  Software Engineer    5.9            4               
  Physical Activity Level Stress Level BMI Category Blood Pressure Heart Rate
1 42                      6            Overweight   126/83         77        
2 60                      8            Normal       125/80         75        
3 60                      8            Normal       125/80         75        
4 30                      8            Obese        140/90         85        
5 30                      8            Obese        140/90         85        
6 30                      8            Obese        140/90         85        
  Daily Steps Sleep Disorder
1  4200       None          
2 10000       None          
3 10000       None          
4  3000       Sleep Apnea   
5  3000       Sleep Apnea   
6  3000       Insomnia

**Description:** \
The Sleep Health and Lifestyle Dataset consists of 374 observations and 13 variables related to individuals' sleep habits, physical health, and lifestyle choices. It aims to analyze the factors influencing sleep quality and overall well-being. This dataset is clarified as synthetic and created by Laksika Tharmalingam for illustrative purposes, (thus its not observational nor experimantal) and can be retrieved through Kaggle. (https://www.kaggle.com/datasets/uom190346a/sleep-health-and-lifestyle-dataset?resource=download)

**Variables:** 
|Variable Name|Type|Description|
|-------------|----|-----------|
|`Person ID`|Integer|An identifier for each individual.|
|`Gender`|Categorical|The gender of the person (Male/Female).|
|`Age`|Integer|The age of the person in years.|
|`Occupation`|String|The occupation or profession of the person.|
|`Sleep Duration`|Float|The number of hours the person sleeps per day.|
|`Quality of Sleep`|Integer|A subjective rating of the quality of sleep, ranging from 1 to 10.|
|`Physical Activity Level`|Integer|The number of minutes the person engages in physical activity daily.|
|`Stress Level`|Integer|A subjective rating of the stress level experienced by the person, ranging from 1 to 10.|
|`BMI Category`|Categorical|The BMI category of the person (Underweight, Normal, Overweight).|
|`Blood Pressure`|String|The blood pressure measurement of the person, indicated as systolic pressure over diastolic pressure. (stored as a character)|
|`Heart Rate`|Integer|The resting heart rate of the person in beats per minute.|
|`Daily Steps`|Integer|The number of steps the person takes per day.|
|`Sleep Disorder`|Categorical|The presence or absence of a sleep disorder in the person (None, Insomnia, Sleep Apnea).|

**Pre-Selection Variables** \
`Person ID` can be dropped since they do not provide sufficient information about sleep and lifestyle information. 

### b) Explanatory Data Analysis

In [3]:
# convert the variable names into snake_case format
colnames(sleep_data) <- gsub(" ", "_", tolower(colnames(sleep_data)))

# split the blood_pressure variable into systolic_bp and diastolic_bp and mutate into a numerical format
sleep_data <- sleep_data |>
separate(blood_pressure, into = c("systolic_bp", "diastolic_bp")) |>
mutate(systolic_bp = as.numeric(systolic_bp),
       diastolic_bp = as.numeric(diastolic_bp))

# mutate bmi_category into "Underweight", "Normal", "Overweight", and "Obese"
# I chose to keep "Obese" since it's recognized by the governemnt of Canada as one of the classification of BMI
sleep_data <- sleep_data |>
mutate(bmi_category = if_else(bmi_category %in% c("Normal", "Normal Weight"), "Normal", bmi_category))


head(sleep_data)

person_id gender age occupation           sleep_duration quality_of_sleep
1 1         Male   27  Software Engineer    6.1            6               
2 2         Male   28  Doctor               6.2            6               
3 3         Male   28  Doctor               6.2            6               
4 4         Male   28  Sales Representative 5.9            4               
5 5         Male   28  Sales Representative 5.9            4               
6 6         Male   28  Software Engineer    5.9            4               
  physical_activity_level stress_level bmi_category systolic_bp diastolic_bp
1 42                      6            Overweight   126         83          
2 60                      8            Normal       125         80          
3 60                      8            Normal       125         80          
4 30                      8            Obese        140         90          
5 30                      8            Obese        140         90          
6 30                      8            Obese        140         90          
  heart_rate daily_steps sleep_disorder
1 77          4200       None          
2 75         10000       None          
3 75         10000       None          
4 85          3000       Sleep Apnea   
5 85          3000       Sleep Apnea   
6 85          3000       Insomnia

### c) Methods: Plan

## Discussion

## References